src/main/java/net/fabricmc/loom/configuration/providers/minecraft/assets/MinecraftAssetsProvider.java

```diff
@@ -99,14 +117,24 @@ public static void provide(MinecraftProvider minecraftProvider, Project project)
 			String filename = "objects" + File.separator + sha1.substring(0, 2) + File.separator + sha1;
 			File file = new File(assets, filename);
 
-			if (!file.exists() || !Checksum.equals(file, sha1)) {
+			String localFileChecksum = !file.exists() ? null : checksumInfos.computeIfAbsent(entry.getKey(), path -> {
+				try {
+					return Files.asByteSource(file).hash(Hashing.sha1()).toString();
+				} catch (IOException e) {
+					e.printStackTrace();
+					return null;
+				}
+			});
+
+			if (localFileChecksum == null || !localFileChecksum.equals(sha1)) {
```

Comment:
This should respect respect --refresh-dependencies

Reply:
I don't think previously the asset downloading respect refresh dependencies, but I have added that now.